manufacturing-client/src/main.rs

```diff
@@ -325,10 +327,15 @@ async fn main() -> Result<()> {
 
                 mfg_string_type = args.mfg_string_type;
                 // ensure that we are given the iface if MACAddress is selected
-                if mfg_string_type == MfgStringType::MACAddress && args.iface.is_none() {
-                    bail!("When using MACAddress as the MfgStringType --iface is required");
+                if mfg_string_type == MfgStringType::MACAddress && args.iface.is_some() {
+                    iface = args.iface;
+                } else {
+                    // If user has not selected any specific iface then default iface will be used
+                    iface = get_default_network_iface()?;
```

Comment:
we can also end up in this `else` part if `mfg_string_type == MfgStringType::SeriaNumber`, so  we don't need to get the default iface in that case